packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -497,8 +459,45 @@ export class GCFBootstrapper {
     // The payload from the scheduler is updated with additional information
     // providing context about the organization/repo that the event is
     // firing for.
+    const payload = {
+      ...body,
+      ...this.buildRepositoryDetails(repoFullName),
+    };
+    try {
+      await this.enqueueTask({
+        id,
+        name: eventName,
+        signature,
+        body: JSON.stringify(payload),
+      });
+    } catch (err) {
+      logger.warn(err.message);
+    }
+  }
+
+  private parsePubSubPayload(req: express.Request) {
+    const body = this.parseRequestBody(req);
+    return {
+      ...body,
+      ...(body.repo ? this.buildRepositoryDetails(body.repo) : {}),
+    };
+  }
+
+  private parseRequestBody(req: express.Request): Scheduled {
+    let body = (Buffer.isBuffer(req.body)
+      ? JSON.parse(req.body.toString('utf8'))
+      : req.body) as Scheduled;
+    // PubSub messages have their payload encoded in body.message.data
+    // as a base64 blob.
+    if (body.message && body.message.data) {
+      body = JSON.parse(Buffer.from(body.message.data, 'base64').toString());
```

Comment:
Perhaps add a TODO message to add a test for this.

Reply:
Yeah, our tests should have caught my oversight. I'll add that in.